### PR TITLE
Fix problems encountered trying to picosvg Noto region flags

### LINKED ISSUE #179: parse_svg_transform fails if multiple transforms are not separated by comma or space
E.g. this EU flag:
https://github.com/googlefonts/noto-emoji/blob/master/third_party/region-flags/svg/EU.svg

contains stuff like `transform="translate(0,-1)rotate(18)"`, which makes picosvg choke with this error:

```
$ picosvg /Users/clupo/Github/noto-emoji/third_party/region-flags/svg/EU.svg
Traceback (most recent call last):
  File "/Users/clupo/Github/warp/.venv/bin/picosvg", line 8, in <module>
    sys.exit(main())
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/picosvg.py", line 66, in main
    app.run(_run, argv=argv)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/absl/app.py", line 303, in run
    _run_main(main, args)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/picosvg.py", line 47, in _run
    svg = SVG.parse(input_file).topicosvg()
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg.py", line 945, in topicosvg
    svg.topicosvg(inplace=True)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg.py", line 955, in topicosvg
    self.ungroup(inplace=True)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg.py", line 513, in ungroup
    self._ungroup(self.svg_root)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg.py", line 450, in _ungroup
    self._inherit_group_attrib(group, child)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg.py", line 417, in _inherit_group_attrib
    attrib_handlers[attr_name](attrib, child, attr_name)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg.py", line 384, in _inherit_matrix_multiply
    transform = Affine2D.fromstring(child.attrib[attr_name])
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg_transform.py", line 67, in fromstring
    return parse_svg_transform(raw_transform)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg_transform.py", line 208, in parse_svg_transform
    args = [float(p) for p in re.split(r"\s*[,\s]\s*", match.group(2))]
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg_transform.py", line 208, in <listcomp>
    args = [float(p) for p in re.split(r"\s*[,\s]\s*", match.group(2))]
ValueError: could not convert string to float: '-1)rotate(18'
```

### LINKED ISSUE #180: when viewBox is missing, picosvg should use width/height to resolve gradient units relative to viewport
this flag emoji has no viewBox, only width/height attributes:
https://github.com/googlefonts/noto-emoji/blob/master/third_party/region-flags/svg/FK.svg

The SVG file contains a gradient with `gradientUnits="userSpaceOnUse"` and implicit coordinates, which default to relative percentages of the viewport:

```xml
<linearGradient id="gradbase" gradientUnits="userSpaceOnUse">
<stop stop-color="#a43907" stop-opacity="0.9961" offset="0"/>
<stop stop-color="#fff" offset="1"/>
</linearGradient>
```

Currently we assume a viewBox is always present and take the width and height of the viewport (for resolving relative coords/lengths) from the viewBox. But if the latter is missing, picosvg fails with this error if it encounters a gradient like the one above:

```
$ picosvg /Users/clupo/Github/noto-emoji/third_party/region-flags/svg/FK.svg
Traceback (most recent call last):
  File "/Users/clupo/Github/warp/.venv/bin/picosvg", line 8, in <module>
    sys.exit(main())
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/picosvg.py", line 66, in main
    app.run(_run, argv=argv)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/absl/app.py", line 303, in run
    _run_main(main, args)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/picosvg.py", line 47, in _run
    svg = SVG.parse(input_file).topicosvg()
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg.py", line 945, in topicosvg
    svg.topicosvg(inplace=True)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg.py", line 966, in topicosvg
    self._apply_gradient_translation(inplace=True)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg.py", line 830, in _apply_gradient_translation
    gradient = _GRADIENT_CLASSES[strip_ns(el.tag)].from_element(
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg_types.py", line 800, in from_element
    width, height = _parse_common_gradient_parts(self, el, view_box)
  File "/Users/clupo/Github/warp/.venv/lib/python3.9/site-packages/picosvg/svg_types.py", line 782, in _parse_common_gradient_parts
    return self.gradientUnits.w, self.gradientUnits.h
AttributeError: 'NoneType' object has no attribute 'w'
```

### LINKED ISSUE #182: picosvg fails if a <desc> is present

### LINKED ISSUE #185: stroke-dasharray none kerplodes
none is the default but svg_types sets "". If user declares none kaboom:

```
  File "src/picosvg/svg_types.py", line 244, in stroke_commands
    dash_array = [float(v) for v in re.split(r"[, ]", self.stroke_dasharray) if v]
  File "src/picosvg/svg_types.py", line 244, in <listcomp>
    dash_array = [float(v) for v in re.split(r"[, ]", self.stroke_dasharray) if v]
ValueError: could not convert string to float: 'none'
```

### LINKED ISSUE #186: Non-int viewbox kerplodes

### LINKED ISSUE #187: Inkscape noise kerplodes
picosvg fails on files like https://raw.githubusercontent.com/googlefonts/noto-emoji/master/third_party/region-flags/svg/AF.svg; perhaps we should just destroy all non-svg (or xlink) content

### PR DESCRIPTION
**Pull request by rsheeter**
Fixes #179 
Fixes #182 
Fixes #180
Fixes #185 
Fixes #186
Fixes #187

### CODE REVIEW

**Comment by yisibl** [comment]
Do we consider introducing an svg compressor, otherwise there will always be a lot of edge cases to deal with.

e.g. 
* svgcleaner: https://github.com/RazrFalcon/svgcleaner/blob/master/docs/svgcleaner.adoc#remove-title-element
* scour: https://github.com/scour-project/scour/pull/102

File: src/picosvg/svg_transform.py [hunk 1]
```diff
@@ -194,16 +194,12 @@ def _fix_rotate(args):
...
-    svg_transforms = re.split(r"(?<=[)])\s*[,\s]\s*(?=\w)", raw_transform)
-    for svg_transform in svg_transforms:
-        match = re.match(
-            r"(?i)(matrix|translate|scale|rotate|skewX|skewY)\((.*)\)", svg_transform
-        )
-        if not match:
-            raise ValueError(f"Unable to parse {raw_transform}")
-
+    for match in re.finditer(
+        r"(?i)(matrix|translate|scale|rotate|skewX|skewY)\s*\(([^)]*)\)", raw_transform
```
Comment (anthrotype) [inline src/picosvg/svg_transform.py:201]: this change to parse_svg_transform looks good but is not captured in the commit message

File: src/picosvg/svg_meta.py [hunk 2]
```diff
@@ -36,10 +36,15 @@ def xlinkns():
     return "http://www.w3.org/1999/xlink"
 
 
+def splitns(name):
+    if "}" in name:
```
Comment (anthrotype) [inline src/picosvg/svg_meta.py:40]: I guess you know lxml has the QName class for qualified XML names, with which you can split the `namespace` and `localname` attributes:
https://lxml.de/api/lxml.etree.QName-class.html

should we use that here?

File: src/picosvg/svg.py [hunk 3]
```diff
@@ -739,6 +740,48 @@ def remove_unpainted_shapes(self, inplace=False):
...
+            if ns not in good_ns:
+                el_to_rm.append(el)
+                continue
+            for attr in el.attrib:
+                ns, _ = splitns(attr)
+                if ns not in good_ns:
+                    attr_to_rm.append(attr)
+            for attr in attr_to_rm:
+                del el.attrib[attr]
+                print("del", attr)
```
Comment (anthrotype) [inline src/picosvg/svg.py:768]: you forgot a `print`

**Review by anthrotype — APPROVED** [review]
LGTM, with a couple comments

**Comment by rsheeter** [comment]
> Are we considering introducing an svg compressor?

Not currently planning to; I imagine that being left to the user to include in their pipeline ahead of picosvg or nanoemoji if they see fit.

File: src/picosvg/svg_meta.py (hunk 2 above)
Comment (rsheeter) [reply src/picosvg/svg_meta.py:40]: I always forget that exists!

File: src/picosvg/svg.py (hunk 3 above)
Comment (rsheeter) [reply src/picosvg/svg.py:768]: ty

File: src/picosvg/svg_transform.py (hunk 1 above)
Comment (rsheeter) [reply src/picosvg/svg_transform.py:201]: Oops ty as always :)

**Review by anthrotype — APPROVED** [review]

**Pull request merged**